awx/main/models/unified_jobs.py

```diff
@@ -1512,3 +1514,47 @@ def log_lifecycle(self, state, blocked_by=None):
         else:
             msg = f"{self._meta.model_name}-{self.id} {state.replace('_', ' ')}"
         logger_job_lifecycle.debug(msg, extra=extra)
+
+#not doing the recursive logic right now , but just trying to make sure it returns something , the logic is basically what is used in the UI.
+    @property
+    def launched_by(self):
+        
+        #sdb.set_trace()
+        #checking if the current job was started by a workflow job.
+       
+        ancestor_job=self.ancestor_job
+
+        if hasattr(ancestor_job,'launch_type'):
+            if ancestor_job.launch_type=="manual":
+                return ancestor_job.created_by
+            elif ancestor_job.launch_type=="scheduled":
+                return ancestor_job.schedule
+            elif ancestor_job.launch_type=="dependency":
+                if ancestor_job.job_type_name=="project_update":
+                    return ancestor_job.project
+                elif ancestor_job.job_type_name=="inventory_update":
+                    return ancestor_job.inventory
+            else:
+                return None
+        else:
+            return None
+
+    @property
+    def ancestor_job(self):
+        
+
+        if hasattr(self,'get_workflow_job'):
```

Comment:
Similarly for this check:

    >>> Counter(hasattr(j, 'get_workflow_job') for j in UnifiedJob.objects.all())
    Counter({True: 7923})